### PR TITLE
[FLINK-21862] Add a Java SDK Showcase tutorial

### PR DESCRIPTION
**Pull request by tzulitai**
This Java project is intended for new StateFun users that would like to start implementing their StateFun application functions using Java (or any other JVM language).

The tutorial is streamlined and split into a few parts which we recommend users to go through a specific order.

Each part is demonstrated with some code snippets plus Javadocs and comments to guide users through the SDK fundamentals.

Users can import this project into an IDE (we suggest IntelliJ IDEA) to ease navigation through the classes and pointers throughout the tutorial.

Please see the README.md on more details of how this showcase tutorial is lay out:
https://github.com/apache/flink-statefun-playground/pull/3/files#diff-3939e8c8eb6416ca78adfcdf15f7e158a829f68f716ca0086b4f49ae126cbc15

### CODE REVIEW

**Review by igalshilman — APPROVED** [review]
Thanks @tzulitai !

I have built it locally (`jdk11`) and ran the examples, and it works 🥳 

I haven't went trough every single line of this PR, but I do recognize all the stuff from our offiline discussions, so really LGTM!

Great job with the JavaDocs, I think they drive very well the flow!

File: java/showcase/pom.xml [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+        <dependency>
+            <groupId>com.google.protobuf</groupId>
+            <artifactId>protobuf-java</artifactId>
+            <version>3.11.4</version>
+        </dependency>
+
+        <!-- Undertow web server -->
+        <dependency>
+            <groupId>io.undertow</groupId>
+            <artifactId>undertow-servlet</artifactId>
```
[the remark this answers is not in the document]
Comment (tzulitai) [reply java/showcase/pom.xml:58]: Updated to use core

File: java/showcase/src/main/java/org/apache/flink/statefun/playground/java/showcase/part1/types/my-protobuf-type.proto [hunk 2]
```diff
@@ -0,0 +1,33 @@
+/*
```
[the remark this answers is not in the document]
Comment (tzulitai) [reply java/showcase/src/main/java/org/apache/flink/statefun/playground/java/showcase/part1/types/my-protobuf-type.proto:1]: Moved to src/main/protobuf

**Comment by tzulitai** [comment]
Thanks for the review, merging this.

**Pull request closed without merging**